src/algebra/continued_fractions/computation/approximations.lean

```diff
@@ -297,18 +303,299 @@ begin
-                            ... ≤ (gcf.of v).denominators (n + 1) : le_of_succ_nth_denom
-                                                                      nth_part_denom_eq }
+      g.denominators n ≤ b * g.denominators n   : by simpa using (mul_le_mul_of_nonneg_right
+                                                                 this zero_le_of_denom)
+                   ... ≤ g.denominators (n + 1) : le_of_succ_nth_denom nth_part_denom_eq }
+end
+
+/-
+Next we prove the so-called *determinant formula* for `gcf.of`:
+`Aₙ * Bₙ₊₁ - Bₙ * Aₙ₊₁ = (-1)^(n + 1)`.
+-/
+section determinant
+
+lemma determinant_aux (hyp: n = 0 ∨ ¬(gcf.of v).terminated_at (n - 1)) :
+  let g := gcf.of v in
+  (g.continuants_aux n).a * (g.continuants_aux (n + 1)).b -
+  (g.continuants_aux n).b * (g.continuants_aux (n + 1)).a = (-1)^n :=
+begin
+  induction n with n IH,
+  case nat.zero { simp [continuants_aux] },
+  case nat.succ
+  { -- set up some shorthand notation
+    let g := gcf.of v,
+    let conts := continuants_aux g (n + 2),
+    set pred_conts := continuants_aux g (n + 1) with pred_conts_eq,
+    set ppred_conts := continuants_aux g n with ppred_conts_eq,
+    let pA := pred_conts.a,
+    let pB := pred_conts.b,
+    let ppA := ppred_conts.a,
+    let ppB := ppred_conts.b,
+    -- let's change the goal to something more readable
+    change pA * conts.b - pB * conts.a = (-1)^(n + 1),
+    have not_terminated_at_n : ¬terminated_at g n, from or.resolve_left hyp n.succ_ne_zero,
+    obtain ⟨gp, s_nth_eq⟩ : ∃ gp, g.s.nth n = some gp, from
+      option.ne_none_iff_exists'.elim_left not_terminated_at_n,
+    -- unfold the recurrence relation for `conts` once and simplify to derive the following
+    suffices : pA * (ppB + gp.b * pB) - pB * (ppA + gp.b * pA) = (-1)^(n + 1), by
+    { simp only [conts, (continuants_aux_recurrence s_nth_eq ppred_conts_eq pred_conts_eq)],
+      have gp_a_eq_one : gp.a = 1, from of_part_num_eq_one (part_num_eq_s_a s_nth_eq),
+      rw [gp_a_eq_one, this.symm],
+      ring },
+    suffices : pA * ppB - pB * ppA = (-1)^(n + 1), calc
+      pA * (ppB + gp.b * pB) - pB * (ppA + gp.b * pA)
+          = pA * ppB + pA * gp.b * pB - pB * ppA - pB * gp.b * pA : by ring
+      ... = pA * ppB - pB * ppA                                   : by ring
+      ... = (-1)^(n + 1)                                          : by assumption,
+    suffices : ppA * pB - ppB * pA = (-1)^n, by
+    { have pow_succ_n : (-1 : K)^(n + 1) = (-1) * (-1)^n, from pow_succ (-1) n,
+      rw [pow_succ_n, ←this],
+      ring },
+    exact (IH $ or.inr $ mt (terminated_stable $ n.sub_le 1) not_terminated_at_n) }
+end
+
+/-- The determinant formula `Aₙ * Bₙ₊₁ - Bₙ * Aₙ₊₁ = (-1)^(n + 1)` -/
+lemma determinant (not_terminated_at_n : ¬(gcf.of v).terminated_at n) :
+  let g := gcf.of v in
+  g.numerators n * g.denominators (n + 1) - g.denominators n * g.numerators (n + 1)
+  = (-1)^(n + 1) :=
+(determinant_aux $ or.inr $ not_terminated_at_n)
+
+end determinant
+
+/-
```

Comment:
```suggestion
/-!
```
This turns the comment block into a module docstring, which will show up in the online mathlib docs.
I think it makes sense to move this module docstring inside the section it belongs to.

Reply:
Changed it where applicable here and other places in `continued_fractions`.